src/ORM/Marshaller.php

```diff
@@ -410,13 +416,13 @@ protected function _belongsToMany(BelongsToMany $assoc, array $data, array $opti
                     }
 
                     if ($forceNew && !$target->exists($rowConditions)) {
-                        $records[$i] = $this->one($row, $options);
+                        $records[$i] = $this->one($row, $options + ['junctionProperty' => $junctionProperty]);
```

Comment:
Instead of making new arrays for each loop iteration, should we create one copy of the modified options once before the loop?

Does the merge() call on 454 need `junctionProperty` as well?